### PR TITLE
Add typescript

### PR DESCRIPTION
**Pull request by theBowja**

### CODE REVIEW

**Comment by theBowja** [comment]
@braineo Can you help test if the index.d.ts works? Otherwise I'll just merge to main and call it done.

**Review by braineo — COMMENTED** [review]
Awesome! I ran some simple checks. It looks pretty good. Just something minor.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,449 @@
+export interface QueryFunction<R> {
+    <Q extends string, O extends QueryOptions = {}>(query: Q, opts?: O):
```
[the remark this answers is not in the document]
Comment (theBowja) [reply index.d.ts:2]: Would it still work if I just remove the `= {}`?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+        fandom: string;
+    };
+    stats: StatFunction;
+}
+
+//#endregion
+
+//#region WeaponMaterial
+
+export interface WeaponMaterial { // English only
```
[the remark this answers is not in the document]
Comment (theBowja) [reply index.d.ts:437]: fixed with c290cf7

File: index.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (braineo) [reply index.d.ts:2]: I think that would actually do

[the remark this answers is not in the document]
Comment (theBowja) [reply index.d.ts]: 42c9c0e

**Pull request merged**